app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/uploadsupportingdocs/UploadSupportingEvidenceController.scala

```diff
@@ -0,0 +1,1012 @@
+            ),
+          newDoYouWantToUploadSupportingEvidenceAnswer =>
+            if (newDoYouWantToUploadSupportingEvidenceAnswer) {
+              val updatedAnswers = answers match {
+                case IncompleteUploadSupportingEvidenceAnswers(None, supportingEvidences) =>
+                  IncompleteUploadSupportingEvidenceAnswers(
+                    Some(newDoYouWantToUploadSupportingEvidenceAnswer),
+                    supportingEvidences
+                  )
+                case IncompleteUploadSupportingEvidenceAnswers(
+                    Some(oldDoYouWantToUploadSupportingEvidenceAnswer),
+                    supportingEvidences
+                    ) =>
+                  if (newDoYouWantToUploadSupportingEvidenceAnswer =!= oldDoYouWantToUploadSupportingEvidenceAnswer) {
+                    IncompleteUploadSupportingEvidenceAnswers(
+                      Some(newDoYouWantToUploadSupportingEvidenceAnswer),
+                      List.empty
+                    )
+                  } else {
+                    IncompleteUploadSupportingEvidenceAnswers(
+                      Some(newDoYouWantToUploadSupportingEvidenceAnswer),
+                      supportingEvidences
+                    )
+                  }
+                case CompleteUploadSupportingEvidenceAnswers(
+                    oldDoYouWantToUploadSupportingEvidenceAnswer,
+                    supportingEvidences
+                    ) =>
+                  if (newDoYouWantToUploadSupportingEvidenceAnswer =!= oldDoYouWantToUploadSupportingEvidenceAnswer) {
+                    IncompleteUploadSupportingEvidenceAnswers(
+                      Some(newDoYouWantToUploadSupportingEvidenceAnswer),
+                      List.empty
+                    )
+                  } else {
+                    IncompleteUploadSupportingEvidenceAnswers(
+                      Some(newDoYouWantToUploadSupportingEvidenceAnswer),
+                      supportingEvidences
+                    )
+                  }
+              }
+
+              val newDraftReturn = fillingOutReturn.draftReturn match {
+                case s @ DraftSingleDisposalReturn(
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _
+                    ) =>
+                  s.copy(uploadSupportingDocuments = Some(updatedAnswers))
+                case m @ DraftMultipleDisposalsReturn(
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _
+                    ) =>
+                  m.copy(uploadSupportingDocuments = Some(updatedAnswers))
+              }
+
+              val result = for {
+                _ <- returnsService
```

Comment:
there's no point calling the BE if the answers haven't changed. If you know the answers haven't changed you can redirect straight to the CYA endpoint without worrying about getting to this for-comprehension. 

If you compare to the submitted answer to the one in session first, you can skip some unnecessary calls here and below.

Reply:
Yes, good shout. Will be addressed after 6th April